src/java/org/apache/cassandra/index/sai/metrics/ColumnQueryMetrics.java

```diff
@@ -70,6 +70,61 @@ public QueryEventListener.PostingListEventListener postingListEventListener()
         }
     }
 
+    public static class BKDIndexMetrics extends ColumnQueryMetrics implements QueryEventListener.BKDIndexEventListener
+    {
+        private static final String BKD_POSTINGS_TYPE = "KDTreePostings";
+
+        /**
+         * BKD index metrics.
+         */
+        private final Timer intersectionLatency;
+        private final Meter postingsNumPostings;
+        private final Meter intersectionEarlyExits;
```

Comment:
nit: Although if we are feeling in the mood to document, field JavaDoc on these three might be helpful. `postingsNumPostings`, for example, might be a little opaque without it.

Reply:
Maybe we could just refer to `/** see {@link QueryEventListener.BKDIndexEventListener#onIntersectionEarlyExit()} */`